extensions-builtin/extra-options-section/scripts/extra_options_section.py

```diff
@@ -70,7 +71,7 @@ def before_process(self, p, *args):
 """),
     "extra_options_txt2img": shared.OptionInfo([], "Settings for txt2img", ui_components.DropdownMulti, lambda: {"choices": list(shared.opts.data_labels.keys())}).js("info", "settingsHintsShowQuicksettings").info("setting entries that also appear in txt2img interfaces").needs_reload_ui(),
     "extra_options_img2img": shared.OptionInfo([], "Settings for img2img", ui_components.DropdownMulti, lambda: {"choices": list(shared.opts.data_labels.keys())}).js("info", "settingsHintsShowQuicksettings").info("setting entries that also appear in img2img interfaces").needs_reload_ui(),
-    "extra_options_cols": shared.OptionInfo(1, "Number of columns for added settings", gr.Number, {"precision": 0}).needs_reload_ui(),
+    "extra_options_cols": shared.OptionInfo(1, "Number of columns for added settings", gr.Slider, {"step": 1, "minimum": 1, "maximum": 6}).needs_reload_ui(),

```

Comment:
not sure limiting the slider to maximum of 6 is a good idea
if it were me I would set it to something ridiculously high

Reply:
> if it were me I would set it to something ridiculously high

In practice it doesn't make much sense. Suppose the stars aligned that you have a UWQHD 5K monitor, you put 15 values in extra_options and shift resize handle to the right as much as possible. This will display a maximum of 13 columns.

According to [Desktop Screen Resolution Stats](https://gs.statcounter.com/screen-resolution-stats/desktop/worldwide) most users have FHD and 2K monitors. If on a 2K monitor you shift resize handle to the right, you just get 6 columns (provided that extra_options are not less than 6). Which again seems like a rare case to me :)

Of course, it's worth nothing to specify a `"maximum": 100`. But why confuse a user if they end up seeing no more than 4-6 columns most of the time?

5120w
![5120w-1](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/44464226/384c182c-ab59-4408-ad51-1dd6ab6db876)

5120w
![5120w-2](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/44464226/7e490b32-9653-4bff-a11f-ce05cdd0b868)

2560w
![2560w](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/44464226/4e49daf4-687b-4665-9666-53cbf43a88be)

1920w
![1920w](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/44464226/8e78e293-b06a-404e-b8fc-325997c68853)

Reply:
I don't think it will confuse the user
and it doesn't break anything to have a high value, when there's too many elements gradio just wraps

I'm not saying that you should set it to 100 (you're the one who came up with the number)
but as you have shown yourself 6 elemenst is about the ammout that gradio will display on 1440p monintor
this means a maximum value of 6 is too low
(I personally use a 1440p and I know people who have multiple 4k monitors)

this is not even getting into edge cases where people might use different themes / css / and possibly all the methods of changing the layout or stitching two monitors together

since it doesn't hurt if it's too high I don't think setting a value that is low enough that people can reach under certain circumstances is a good idea

if you're afraid of confusing the user, just add a like
`.info("they column number is also limited by the monitor width and element size")`
> need a better wording of that

Reply:
Yeah, `"maximum": 100` I made up for the sake of example :) Maybe value of 20 would be sufficient, what do you think?

Good idea with the .info(), didn't think of that. I'll make edit a bit later.

Reply:
20 shoud be fine for 99.999% of people
the last 0.001% can figure out things themselves